src/mc_control/fsm/states/Grippers.cpp

```diff
@@ -60,14 +78,26 @@ void Grippers::start(Controller & ctl)
 bool Grippers::run(Controller & ctl)
 {
   if(std::all_of(grippers_.begin(), grippers_.end(),
-                 [&ctl](const std::string & g) { return ctl.grippers[g]->targetQ == nullptr; }))
+                 [&ctl](const std::string & g) { return ctl.grippers[g]->complete(); }))
   {
     output("OK");
     return true;
   }
   return false;
 }
 
+void Grippers::teardown(Controller & ctl)
+{
+  if(config_.has("grippers"))
+  {
+    auto grippers = config_("grippers");
+    for(const auto & g : grippers.keys())
+    {
+      ctl.grippers[g]->resetDefaults();
```

Comment:
I think a better approach is to save the safety settings at the start and restore them here. Otherwise an application may change the default settings and those will get overriden after a Gripper state completes